packages/common/src/ui/components/steppers/VerticalStepper/VerticalStepper.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { FC } from 'react';
+import Box from '@mui/material/Box';
+import Stepper from '@mui/material/Stepper';
+import Step from '@mui/material/Step';
+import StepLabel from '@mui/material/StepLabel';
+import Typography from '@mui/material/Typography';
+import StepConnector, {
+  stepConnectorClasses,
+} from '@mui/material/StepConnector';
+import { styled } from '@mui/material';
+
+import { LocaleKey, useTranslation } from '../../../../intl';
+
+interface StepDefinition {
+  label: LocaleKey;
+  description: string;
+}
+
+interface StepperProps {
+  activeStep: number;
+  steps: StepDefinition[];
+}
+
+const StyledConnector = styled(StepConnector)(({ theme }) => ({
+  '&.MuiStepConnector-root': {
+    // This margin is an adjustment as the icon we use: the circle, is smaller
+    // than what is the default size, so this tries to centre align the connector by
+    // overriding the default.
+    marginLeft: '3px',
+  },
+  [`& .${stepConnectorClasses.line}`]: {
+    borderColor: theme.palette.secondary.light,
+
+    // The width is for the connector which is significantly thicker than the default.
+    // Additionally the height has been shrunk so that each connector hits the icons
+    // on the top and bottom sufficiently.
+    borderWidth: 6,
+    minHeight: 12,
+  },
+  [`&.${stepConnectorClasses.active}`]: {
+    [`& .${stepConnectorClasses.line}`]: {
+      borderColor: theme.palette.secondary.dark,
+    },
+  },
+
+  [`&.${stepConnectorClasses.completed}`]: {
+    [`& .${stepConnectorClasses.line}`]: {
+      borderColor: theme.palette.secondary.dark,
+    },
+  },
+}));
+
+const Circle = styled('div', {
+  shouldForwardProp: prop =>
+    prop !== 'completed' && prop !== 'active' && prop !== 'error',
+})<{
+  completed: boolean;
+  active: boolean;
+}>(({ active, completed, theme }) => {
+  // Base colours for all uncompleted and non-active steps.
+  const colors = {
+    borderColor: theme.palette.secondary.light,
+    backgroundColor: theme.palette.secondary.light,
+  };
+
+  // If the step is completed, then everything is dark.
+  if (completed) {
+    colors.backgroundColor = theme.palette.secondary.dark;
+    colors.borderColor = theme.palette.secondary.dark;
+  }
+
+  // If the step is active, the center of the circle is white
+  if (active) {
+    colors.backgroundColor = 'white';
+  }
+
+  return {
+    border: 'solid',
+
+    // These numbers are arbitrary and are just what look good to me.
+    // our designs have the circle icon at 16px - with the border I
+    // believe this is 15px.. either 4px on the border or 13px on the
+    // dimensions and it looks a bit off.. maybe this isn't how it works.
+    borderWidth: '3px',
+    width: '12px',
+    height: '12px',
+    borderRadius: '16px',
+    ...colors,
+  };
+});
+
+export const VerticalStepper: FC<StepperProps> = ({ activeStep, steps }) => {
```

Comment:
I'm not sure if `VerticalStepper` is good - might be a bit too specific to the use case